wgengine/magicsock/magicsock_linux.go

```diff
@@ -213,6 +213,37 @@ func (c *Conn) receiveDisco(pc net.PacketConn) {
 			// Too small to be a valid UDP datagram, drop.
 			continue
 		}
+
+		dstPort := binary.BigEndian.Uint16(buf[2:4])
+		if dstPort == 0 {
+			c.logf("[unexpected] disco raw: received packet for port 0")
+		}
+
+		var acceptPort uint16
+		if isIPV6 {
+			acceptPort = c.pconn6.Port()
```

Comment:
Looks like `pconn6` can be nil. There are a bunch of checks for it being nil in magicsock. See `initialBind`.

Do `if isIPV6 && p.conn6 != nil` here?

The pconn4 is always non-nil.

Reply:
Good call, I made a new PR https://github.com/tailscale/tailscale/pull/5537 and rebased atop it to remove this condition.